doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1588,17 +1588,26 @@ Controlling the proof flow
       While :tacn:`pose proof` expects that no existential variables are generated by
       the tactic, :tacn:`epose proof` removes this constraint.
 
-.. tacv:: specialize (@ident {* @term}) {? as @simple_intropattern}
-          specialize @ident with @bindings {? as @simple_intropattern}
-   :name: specialize; _
+.. tacn:: specialize @one_term_with_bindings {? @as_ipat }
+
+   .. todo PR: Where is @ident now?  Need to adapt
```

Comment:
`@ident` is the head constant of the `@one_term` provided as argument.

Reply:
OK.  Still quite murky on the high level description of specialize, need to see more examples.  The existing text says "This tactic works on local hypothesis :n:`@ident`."  That seems wrong; it need not be a hypothesis and it makes more sense to talk about the `@one_term` than the head constant.

Suppose the `@one_term` is:

![image](https://user-images.githubusercontent.com/1253341/201776143-c27e0906-a9d7-4a63-a2e2-bfeb65eb4dba.png)

The current description says: "The premises of this hypothesis (either universal quantifications or
non-dependent implications) are instantiated by concrete _terms_ from
:n:`@one_term_with_bindings`."  The premises in this case are the `t__i`?  And for `A -> B -> C`, they are A and B?  How do you get multiple _terms_?